services/storage/src/simcore_service_storage/access_layer.py

```diff
@@ -89,13 +93,26 @@ def __str__(self):
         return "Error in {}: {} [{}]".format(self.identifier, self.reason, self.details)
 
 
+class ProjectNotFoundtError(Exception):
+    def __init__(self, project_id: ProjectID) -> None:
+        super().__init__(f"No project found for {project_id=}")
+        self.project_id = project_id
+
+
 async def _get_user_groups_ids(conn: SAConnection, user_id: int) -> List[int]:
     stmt = sa.select([user_to_groups.c.gid]).where(user_to_groups.c.uid == user_id)
     rows = await (await conn.execute(stmt)).fetchall()
     user_group_ids = [g.gid for g in rows]
     return user_group_ids
 
 
+async def _is_project_present(conn: SAConnection, project_id: ProjectID) -> bool:
```

Comment:
```suggestion
async def _project_exists(conn: SAConnection, project_id: ProjectID) -> bool:
```

Reply:
``_project_exists`` definitively a better name (analogous to ``path.exists()``, etc ... )